client/components/app-promo/index.jsx

```diff
@@ -104,9 +79,27 @@ export default React.createClass( {
 						height="32"
 						alt="WordPress Desktop Icon"
 					/>
-					{ this.state.promo_item.message }
+					{ message }
 				</a>
 			</div>
 		);
 	}
 } );
+
+AppPromo.defaultProps = {
+	promoItem: getRandomPromo(),
+	translate: identity,
+	recordTracksEvent: noop,
+	saveDismissal: () => store.set( 'desktop_promo_disabled', true ),
+	getPromoLink
+};
+
+const mapDispatchToProps = ( dispatch ) => {
+	return {
```

Comment:
This could be simplified using either [bindActionCreators](http://redux.js.org/docs/api/bindActionCreators.html):

```js
const mapDispatchToProps = ( dispatch ) => bindActionCreators( { recordTracksEvent }, dispatch );
```

... or the [object shorthand of `mapDispatchToProps`](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options):

```js
export default connect( null, { recordTracksEvent } )( localize( AppPromo ) ) ;
```